lib/puppet/provider/foreman_smartproxy/rest_v2.rb

```diff
@@ -2,6 +2,12 @@
 
   confine :feature => :apipie_bindings
 
+  def print_warning(e)
+    body = JSON.parse(e.response)["error"]["full_messages"].join(" ") rescue 'N/A'
+    warning("Error while registering proxy (#{e.message}): #{body}, " +
+      "registration can be skipped by unsetting --register-in-foreman option.")
```

Comment:
This message is rather tied to the module.. where the parameter is `register_in_foreman`, or if you meant the foreman-installer option, then it's `--foreman-register-in-foreman`.